presto-main/src/test/java/com/facebook/presto/type/TestEnumLiterals.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.common.type.VarcharEnumType;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+public class TestEnumLiterals
+        extends TestEnumBase
+{
+    @Test
+    public void testLongEnumLiterals()
+    {
+        assertFunction("Mood.HAPPY", MOOD_ENUM, 0L);
```

Comment:
These are not functions, so why are we using `assertFunction` 😂

Reply:
well technically it gets rewritten to a CAST so it's still a function 🙈

Reply:
changing directions for tests – see `TestEnums.java`